st2common/st2common/models/api/pack.py

```diff
@@ -48,67 +49,95 @@ class PackAPI(BaseAPI):
+                'description': 'Brief description of the pack and the service it integrates with.',
                 'required': True
             },
-            'description': {
-                'type': 'string'
-            },
             'keywords': {
                 'type': 'array',
+                'description': 'Keywords describing the pack.',
                 'items': {'type': 'string'},
                 'default': []
             },
             'version': {
                 'type': 'string',
+                'description': 'Pack version. Must follow the semver format '
+                               '(for instance, "0.1.0").',
                 'pattern': PACK_VERSION_REGEX,
                 'required': True
             },
             'author': {
                 'type': 'string',
+                'description': 'Pack author or authors.',
                 'required': True
             },
             'email': {
                 'type': 'string',
-                'format': 'email',
-                'required': True
+                'description': 'E-mail of the pack author.',
+                'format': 'email'
             },
             'files': {
                 'type': 'array',
+                'description': 'A list of files inside the pack.',
+                'items': {'type': 'string'},
+                'default': []
+            },
+            'dependencies': {
+                'type': 'array',
+                'description': 'A list of other StackStorm packs this pack depends upon. '
+                               'The same format as in "st2 pack install" is used: '
+                               '"<name or full URL>[=<version or git ref>]".',
                 'items': {'type': 'string'},
                 'default': []
+            },
+            'system': {
+                'type': 'object',
+                'description': 'Specification for the system components and packages '
+                               'required for the pack.',
+                'properties': {
+                    'stackstorm': {
+                        'type': 'string',
+                        'description': 'Required StackStorm version. Examples: ">1.6", '
+                                       '">=1.8dev, <2.0"',
+                        'pattern': ST2_VERSION_REGEX,
+                    }
+                }
             }
-        },
-        'additionalProperties': False
+        }
     }
 
     @classmethod
     def to_model(cls, pack):
-        name = pack.name
-        description = pack.description
-        ref = pack.ref
-        keywords = getattr(pack, 'keywords', [])
-        version = str(pack.version)
-        author = pack.author
-        email = pack.email
-        files = getattr(pack, 'files', [])
-
-        model = cls.model(name=name, description=description, ref=ref, keywords=keywords,
-                          version=version, author=author, email=email, files=files)
+        parameters = pack.__dict__
```

Comment:
I'm personally not a fan of this __dict__ approach, too much magic and could cause us to store weird stuff.

I would prefer to explicitly assign it as we did before. As far as forward compatibility goes - I thought we will just accept them without throwing, but not actually store them in the database (which I think it's the right thing to do).